### PR TITLE
Fix pg_dump with multiple '-t' arguments

### LINKED ISSUE #9059: multiple '-t' arguments cause `pg_dump` fail with "syntax error at or near UNION"
### Greenplum version or build
master 32cc07cc77a07be30da0ea9d2a6a2d3a498339f9

### OS version and uname -a
Linux cc-desktop 4.19.84-1-lts #1 SMP Tue, 12 Nov 2019 22:19:52 +0000 x86_64 GNU/Linux

### autoconf options used ( config.status --config )
```'--with-perl' '--with-python' '--with-libxml' '--enable-debug' '--enable-cassert' '--disable-orca' '--disable-gpcloud' '--disable-gpfdist' '--prefix=/home/cc/gp/master' 'CC=ccache gcc' 'CXX=ccache g++'
```

### Installation information ( pg_config )
using `gpAux/gpdemo` with `MIRROR=false`

### Expected behavior
Create two table `t1` and `t2`
```
CREATE TABLE t1 (id int);
CREATE TABLE t2 (id int);
```

Dump the schemas with `pg_dump`
```
pg_dump -p 5432 -s -t '"public"."t1"' -t '"public"."t2"' postgres
```

### Actual behavior
`pg_dump` fails with error message:

```
pg_dump: [archiver (db)] query failed: ERROR:  syntax error at or near "UNION"
LINE 1: UNION ALL
        ^
pg_dump: [archiver (db)] query was: UNION ALL
SELECT c.oid
FROM pg_catalog.pg_class c
     LEFT JOIN pg_catalog.pg_namespace n
     ON n.oid OPERATOR(pg_catalog.=) c.relnamespace
WHERE c.relkind OPERATOR(pg_catalog.=) ANY
    (array['r', 'S', 'v', 'm', 'f'])
  AND c.relname ~ '^(t2)$'
  AND n.nspname ~ '^(public)$'
```

### Step to reproduce the behavior

See the expected behavior

### NOTE

`pg_dump` works fine with only one `-t` supplied, like:
```
pg_dump -p 5432 -s -t '"public"."t2"' postgres
```

### PR DESCRIPTION
**Pull request by adam8157**
It's an upstream merge issue, just align with the REL9_6_STABLE here.

Fixes #9059

### CODE REVIEW

**Comment by hlinnaka** [comment]
This reverts the code to the same it was at the point in 9.6 dev cycle that we've merged up to. Removing the "UNION ALL" thing seems correct to me, that was indeed a mismerge. However, the search_path stuff came from the 9.4.X minor version merge, and do have the rest of that patch. I don't think we should revert those changes. We would get them back or again in a later 9.6.X minor version merge, but let's avoid the back-and-forth.

In other words, I think we should only remove the "UNION ALL" thing. By doing that, the code would in sync with the tip of pgsql REL9_6_STABLE branch.

**Comment by adam8157** [comment]
Ah, yes! I thought it was removed after 9.4 and before 9.6, forgot the check the STABLE and master branches.

Updated, thanks.
